ui/src/custom-fetch.ts

```diff
@@ -1,47 +1,47 @@
 const getBody = <T>(c: Response | Request): Promise<T> => {
     const contentType = c.headers.get('content-type');
-  
+
     if (contentType && contentType.includes('application/json')) {
       return c.json();
     }
-  
+
     return c.text() as Promise<T>;
   };
-  
+
   // NOTE: Update just base url
   const getUrl = (contextUrl: string): string => {
-    const baseUrl = VITE_API_URL;
-    const url = new URL(baseUrl + contextUrl);
+    const baseUrl = VITE_API_URL || window.location.origin;
+    const url = new URL(baseUrl + contextUrl, baseUrl);
```

Comment:
First, we calculate the `baseUrl` by checking whether the `VITE_API_URL` env variable was set, otherwise, we fall back to the URL from where the UI is being served.

After that, we pass it to the URL constructor to build the URL considering the `baseUrl`
https://developer.mozilla.org/en-US/docs/Web/API/URL#usage_notes

Reply:
@Croway this took care of both running the UI through `mvn quarkus:dev` and building the CLI locally and running it in `PROD` mode.

Reply:
@orpiske when you have the time, please check it again, especially around serving the router from a different port :pray:

Reply:
Sure thing. I will re-review it today, thanks!

Reply:
sounds good!

Reply:
@lordrip shouldn't it be `new URL(contextUrl, baseUrl)` ?

Reply:
@Croway it works both ways :+1: , but I agree that we're not using it like that, thanks for correcting it :+1: